### PR TITLE
gRPC client auth added

### PR DESCRIPTION
**Pull request by huseyinbabal**
## Description

Changes proposed in this pull request:

- Auth context is added to gRPC client connections

## Testing

Better to test this together with https://github.com/kubeshop/botkube-cloud/pull/431

### CODE REVIEW

**Comment by pkosiec** [comment]
@josefkarasek I've unassigned you from the review as I'll test both server + client changes E2E 👍

**Review by pkosiec — APPROVED** [review]
LGTM 🚀

File: pkg/bot/cloudslack.go [hunk 1]
```diff
@@ -515,3 +524,41 @@ func (b *CloudSlack) getChannelsToNotify(sourceBindings []string) []string {
 	}
 	return out
 }
+
+func (b *CloudSlack) addStreamingClientCredentials() grpc.StreamClientInterceptor {
+	return func(ctx context.Context, desc *grpc.StreamDesc, cc *grpc.ClientConn, method string, streamer grpc.Streamer, opts ...grpc.CallOption) (grpc.ClientStream, error) {
+		remoteCfg, ok := remote.GetConfig()
+		if !ok {
+			return nil, errors.New("while getting remote configuration")
```
Comment (pkosiec) [inline pkg/bot/cloudslack.go:532]:
```suggestion
			return nil, errors.New("empty remote configuration")
```

**Review by pkosiec — APPROVED** [review]
👌

**Comment by huseyinbabal** [comment]
slack e2e failed due to rate limit error, just re-ran the failed test

**Pull request merged**